SO-6369: Fix handling of enums in revision conflict processor (7.x)

AAAlinaaa opened the pull request.

AAAlinaaa committed:
SO-6369: Ensure enums aren't handled as objects

apeteri commented:
Can we get a test case that shows what happens to `Enum` values?

apeteri commented:
The build failed at the Codecov upload so we know that expectations in existing test cases aren't challenged.

AAAlinaaa committed:
SO-6369: Add test for enum property type conflict handling

apeteri reviewed (approved):
👍 

The test case fails with "Couldn't parse json" before the change in `DocumentMapping` is applied, but succeeds afterwards. `isObject` is only called in connection with conflict processing, so it does not affect e.g. mapping for existing indexes.

nagyo commented on commons/com.b2international.index.tests/src/com/b2international/index/revision/RevisionBranchMergeConflictTest.java:
```diff
@@ -100,6 +103,21 @@ public void rebaseContainerDeletedOnTargetAndComponentAddedOnSource() throws Exc
+		EnumPropertyData updateOnBranch = new EnumPropertyData(STORAGE_KEY1, Option.OPTION_A);
+		indexChange(branchA, data, updateOnBranch);
+		
+		branching().prepareMerge(MAIN, branchA).merge();
```
It would be nice to assert the state of the enum after the merge. If I read the code (and not the method name) it's not obvious at first what we are trying to verify here.

apeteri replied:
Unfortunately `assertDoesNotThrow` appears in JUnit 5 only: https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertDoesNotThrow(org.junit.jupiter.api.function.Executable)

The test method breaks with an exception before the fix, and it needs to complete in its entirety to pass.

nagyo replied:
I meant something like:
```
assertEquals(OPTION_A, getRevision(MAIN, data).getOption())
```
so we can make sure that the value does not change as a result of the merge.

nagyo reviewed (approved):
LGTM with a note

apeteri committed:
fix(index.tests): Verify that a consistent enum change on both...

sides can be rebased successfully and the value remains the same
following the rebase.

cmark commented:
Late LGTM!
